### PR TITLE
Only show primary and secondary page links on horizontal nav bar & templatzied horizontal nav code

### LINKED ISSUE #2477: Horizontal Sub Nav should only show "primary" pages
Host FAQ is a child page of Support the Program, but it shouldn't show up in the navigation.
image

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #2477

It would be easier to review this PR commit by commit.

1. made horizontal nav bar show primary and secondary page links (1st commit)
2. templatized horizontal nav code (2nd commit)

---

- campaign page example: https://foundation-mofostaging-pr-2507.herokuapp.com/en/campaigns/multi-page/
- primary page example: https://foundation-mofostaging-pr-2507.herokuapp.com/en/about

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/tags/primary_page_menu.html [hunk 1]
```diff
@@ -7,8 +7,9 @@
       <div class="col-12">
         <div id="multipage-nav">
         	{% for entry in menu_pages %}
-        	  <a class="multipage-link {% if current.title == entry.page.title %}active{% endif %} {% if entry.restriction != None %}multipage-{{entry.restriction}}-restricted{% endif %} {% if not entry.page.live %}multipage-draft{% endif %}" href="{% pageurl entry.page %}">{% if entry.depth == 2 %}→ {% endif %}{{ entry.menu_title }}
-        	  </a>
+          {% if entry.depth < 2 %}
+          <a class="multipage-link {% if current.title == entry.page.title %}active{% endif %} {% if entry.restriction != None %}multipage-{{entry.restriction}}-restricted{% endif %} {% if not entry.page.live %}multipage-draft{% endif %}" href="{% pageurl entry.page %}">{{ entry.menu_title }}</a>
+          {% endif %}
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/tags/primary_page_menu.html]: note: shared code will be templatized in another ticket/PR

Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/tags/primary_page_menu.html]: i lie. i might be able to do it in this PR.

Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/tags/primary_page_menu.html]: k, see my 2nd commit for templatization work

**Comment by mmmavis** [comment]
More nav templatization work coming up as separate PRs.  I didn't wanna complicate this PR too much.

**Comment by Pomax** [comment]
Hm, are https://foundation-mofostaging-pr-2507.herokuapp.com/en/campaigns/multi-page/maintain-space-main/ and https://foundation-mofostaging-pr-2507.herokuapp.com/en/campaigns/multi-page/car-much-it/ supposed to be a different layout than https://foundation-mofostaging-pr-2507.herokuapp.com/en/campaigns/multi-page/visit-eye-dinner/ ?

**Comment by mmmavis** [comment]
@Pomax in `campaign_page.html`... apparently wrapper gets assigned different classname according to the page content it has

```html
<div class="py-4 py-md-5
  {% if page.specific.narrowed_page_content %}
    offset-md-2
    col-md-8
  {% else %}
    col-md-12
  {% endif %}"
>
```

I'm still trying to spot that `narrowed_page_content` toggle on CMS admin panel though.

**Comment by Pomax** [comment]
Ah, I found the toggle: 

image

File: network-api/networkapi/settings.py [hunk 2]
```diff
@@ -266,6 +266,8 @@
                 'homepage_tags': 'networkapi.wagtailpages.templatetags.homepage_tags',
                 'card_tags': 'networkapi.wagtailpages.templatetags.card_tags',
                 'primary_page_tags': 'networkapi.wagtailpages.templatetags.primary_page_tags',
+                'nav_tags':
```
Comment (Pomax) [inline network-api/networkapi/settings.py]: this should fit on one line?

File: network-api/networkapi/settings.py [hunk 3]
```diff
@@ -266,6 +266,8 @@
...
+                    'networkapi.wagtailpages.templatetags.nav_tags',
```
Comment (Pomax) [inline network-api/networkapi/settings.py]: this should fit on one line?

**Comment by Pomax** [comment]
This might be a good time to add a visual regression test, too: we can add a `multi-page-test.js` to the `./cypress/integrations` dir, and make that screenshoot the multipage campaign:

```
describe('Integration test with visual testing', () => {
  it('Loads the homepage', function() {

    // Load the page or perform any other interactions with the app.
    cy.visit(`/en/campaigns/multi-page/`);

    // Take a snapshot for visual diffing
    cy.percySnapshot();
  });
});
```

and now everytime a new PR comes it, the visual regression test will make sure that the multipage layout still looks the same.

**Comment by mmmavis** [comment]
Woohooo~

**Pull request merged**